core/extension/gdextension_interface.h

```diff
@@ -34,6 +34,10 @@
  * Together with the JSON file, you should be able to generate any binder.
  */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
What's the reason for this change? I've googled recommended behavior for this, and GNU warns about having includes in `extern "C"`[^1]:

> Note that putting a #include in an extern "C" { ... } block yields a syntax error in C++ mode on some platforms (e.g., glibc systems with g++ v3.3 to v4.2, AIX, IRIX). For this reason, it is recommended to place the #include before the extern "C" block.

[^1]: https://www.gnu.org/software/gnulib/manual/html_node/Header-files.html